src/plugins/util/components.js

```diff
@@ -31,18 +46,7 @@ const switchButton = (() => {
 
         slider.addEventListener('click', () => {
             input.checked = !input.checked;
```

Comment:
maybe this could use the setState function ? Something like `setState(!input.checked);`

Reply:
i think it's much cleaner to manage internal state directly. The purpose of setState is to expose a way to manage the state, this fn can evolve and this way we avoid side effect. In vue or react framework this is the right way, but i'm open to discussing that, maybe @JulienBolard has an opinion

Reply:
In C/C++, it is a better approche to use the same function everywhere as you might protect access of the state with a lock for example. Not familiar as much with JS but it would make sens here to avoid changeStateRenderer() duplication (and as a pure fictional example send an extra 'stateChange' message somewhere when state changes if needed later). It looks to me a  better approche to keep things centralized in a single fn.